tools/pipeline-witness/Azure.Sdk.Tools.PipelineWitness/RunProcessor.cs

```diff
@@ -63,7 +63,37 @@ public async Task ProcessRunAsync(Uri runUri)
             var projectClient = vssConnection.GetClient<ProjectHttpClient>();
 
             var project = await projectClient.GetProject(projectGuid.ToString());
-            var build = await buildClient.GetBuildAsync(projectGuid, runId);
+
+            // Avoid unhandled BuildNotFoundException by folding it into the following null check.
+            Build build = null;
+
+            try
+            {
+                build = await buildClient.GetBuildAsync(projectGuid, runId);
+            }
+            catch (BuildNotFoundException)
+            {
+            }
+
+            if (build == null)
+            {
+                this.logger.LogWarning("Unable to process run due to missing build");
+                return;
+            }
+
+            if (build.Deleted)
+            {
+                this.logger.LogWarning("Skipping deleted build");
```

Comment:
Not sure if this is a valid property, but in the case that the log context doesn't already add this to the output:

```suggestion
                this.logger.LogWarning($"Skipping deleted build {build.Id}");
```

Reply:
added via structured log message